### ISSUE TITLE
Question about MNIST For ML Beginners, the model definition?

### ISSUE THREAD

**Issue by JacobIsrael123** [tensorflow/tensorflow#620]
https://www.tensorflow.org/versions/master/tutorials/mnist/beginners/index.html#mnist-for-ml-beginners
In the tutorial for beginner, the model definition flipped from original equation, as say it is a small trick to deal with x being a 2D tensor. 
>First, we multiply x by W with the expression tf.matmul(x, W). This is flipped from when we multiplied them in our equation, where we had Wx, as a small trick to deal with x being a 2D tensor with multiple inputs. We then add b, and finally apply tf.nn.softmax.

Why say it is a small trick ?  Both definition should be equivalent right?
As below
Model definition: 
(1) $y = Wx + b$  
Shape: (10, n) = (10, 784) x (784, n) + (10, n)  
(2) $y = xW + b$  
Shape: (n, 10) = (n, 784) x (784, 10) + (n, 10)

I change the downloaded code ```mnist_softmax.py``` code to model (1) version and run it, I found that the accuracy is lower than model (2)?
Anyone could tell me the reason?

**Comment by keveman** [comment]
Hi Jacob,

The 'small trick' refers to the fact that the image tensor returned by `mnist.train.next_batch(100)` has the shape `[100, 784]`. Note that the batch is the innermost dimension. To use it directly, we shape the `W` tensor as `[784, 10]` to use it directly in `tf.matmul(x, W)` to produce `10` activations. So if you transposed the output of `mnist.train.next_batch(100)` to `[784, 100]` and shaped your `W` as `[10, 784]`, then indeed you can call `tf.matmul(W, x)`. Is that what you were trying and were seeing a different accuracy?

**Comment by JacobIsrael123** [comment]
Hello Keveman, 
         Thanks for your reply. My understanding is exactly the same with you. This is exactly what I am trying and seeing than the tensorflow version is always higher than the flipped version. 

Could you also have a try and take a look?


[mnist_softmax_flip.txt](https://github.com/tensorflow/tensorflow/files/75367/mnist_softmax_flip.txt)
[mnist_softmax.txt](https://github.com/tensorflow/tensorflow/files/75368/mnist_softmax.txt)

**Comment by girving** [comment]
This is a question for stackoverflow, not an issue with TensorFlow.  If the activations have shape `[batch, input]` and the weight matrix has shape `[input, output]`, only `tf.matmul(activations, weights)` makes sense.  The other way around would be a size error.  You're correct that transposing the entire graph should produce something with the same semantics, but note that *every* part of the graph needs to be transposed, including the input pipeline, loss ops, etc.
